include/wil/safearrays.h

```diff
@@ -0,0 +1,897 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+#ifndef __WIL_SAFEARRAYS_INCLUDED

```

Comment:
This has an include guard, but then uses the `resource.h`-style guards on definitions. E.g. you wouldn't be able to include this file twice to "turn on" shared/weak pointer support like you can for the `resource.h` header. IMO that's not a huge deal, but we shouldn't be attempting to do both. We should either make it includable multiple times or remove the definition guards (which likely means removing shared/weak support)

Reply:
I'll just remove the outer most guards and leave the two-phase includes, but you shouldn't have to include it twice if you include it last after all the other WIL headers.

Reply:
The issue isn't needing to include it twice, it's avoiding include order dependencies. That is, the following:
```c++
#include <wil/safearray.h>
#include <memory>
#include <wil/safearray.h>
```
Should be equivalent to:
```c++
#include <memory>
#include <wil/safearray.h>
```